src/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs

```diff
@@ -77,25 +77,36 @@ internal AcquireTokenInteractiveParameterBuilder WithCurrentSynchronizationConte
         /// <returns>The builder to chain the .With methods</returns>
         public AcquireTokenInteractiveParameterBuilder WithUseEmbeddedWebView(bool useEmbeddedWebView)
         {
-#if NET_CORE
+#if NET_CORE || NETSTANDARD
             if (useEmbeddedWebView)
             {
-                throw new PlatformNotSupportedException("Only the system browser is available on when using MSAL in a .NET Core context. See See https://aka.ms/msal-net-os-browser for details.");
+                throw new MsalClientException(MsalError.WebviewUnavailable, "An embedded webview is not available on this platform. " +
+                    "Please use WithUseEmbeddedWebView(false) or leave the default. " +
+                    "See https://aka.ms/msal-net-os-browser for details about the system webview.");
+            }
+#elif WINDOWS_APP
+            if (!useEmbeddedWebView)
+            {
+                throw new MsalClientException(
+                   MsalError.WebviewUnavailable,
+                   "On UWP, MSAL does not offer a system webview out of the box. Please set .WithUseEmbeddedWebview to true or leave the default. " +
+                   "To use the UWP Web Authentication Manager (WAM) see https://aka.ms/msal-net-uwp-wam");
             }
 #endif
+
             CommonParameters.AddApiTelemetryFeature(ApiTelemetryFeature.WithEmbeddedWebView);
-            Parameters.UseEmbeddedWebView = useEmbeddedWebView;
+            Parameters.UseEmbeddedWebView = new Maybe<bool>(useEmbeddedWebView);
```

Comment:
>Maybe [](start = 48, length = 5)

do we need the maybe?

Reply:
I'm happy to discuss alterntives, but we really need a 3 state: 

- user not specified
- user specified true & user specified false

We cannot use nullables (bool?) because they are not available on .net45. So I figured we can use the functional Maybe construct which has other benefits as well (when using Linq).

Reply:
Removed and replaced with an enum